### ISSUE TITLE
Wrong Typing Error

### ISSUE THREAD

**Issue by cojok** [microsoft/pyright#834]
**Describe the bug**
I am trying to use pyright VS Code plugin in order to live validate my types in python projects. I am noticing that the plugin is not really working correctly for the env variables, with os.environ or os.getenv(key, default). I am getting the port for db and the linter(pyright) complains about this:

```
Argument of type "str" cannot be assigned to parameter "port" of type "int" in function "__init__"
  "str" is incompatible with "int"Pyright (reportGeneralTypeIssues)
```

The problem here is that os.getenv(key, value) returns a string. I am trying to get the variable type to string but it still complains. I am not sure if this is a wanted behaviour of the plugin or not, but it makes it a bit not usable.

**To Reproduce**
Simple:

.env
```
PORT: 6547
```

file.py
```python
import os

port: str = os.getenv('PORT', 6547);
```

**Expected behavior**
I expect that the linter(pyright) to just validate that the output of the method is the same as the expected python variable type.

**VS Code extension or command-line**
I am using the VS Code extension v1.1.52
